src/state/utils/store_io.ts

```diff
@@ -122,7 +122,7 @@ export const serializedDataToUrl = (data: SerializedStoreData): string => {
  * Loads the viewer state from the given URL parameters. Note that this MUST be
  * called after the collection and dataset are loaded and set in the store.
  */
-export const loadViewerStateFromParams = async (store: Store<ViewerStore>, params: URLSearchParams): Promise<void> => {
+export const loadViewerStateFromParams = (store: Store<ViewerStore>, params: URLSearchParams): void => {
```

Comment:
This function wasn't doing anything that required it to be async, and making it synchronous simplified the error handling.